src/Client.php

```diff
@@ -25,295 +26,233 @@ class Client
+    public function request(string $method, string $url, array $data = []) {
+        if ($method === "POST")
+            $data = [GuzzleHttp\RequestOptions::JSON => $data];
+        $data["http_errors"] = false;
+        var_dump($data);
+        $response = $this->guzzleClient->request($method, $url, $data);
+        $this->checkForError($response);
+        $data = json_decode($response->getBody(), true)["data"];
+        return $data;
     }
 
-    /**
-     * Get list of formatted names from Lassie
-     */
-    public function getListOfNames($active = false)
+    /** Get name corresponding to an ID in Lassie */
+    public function getNameByID($id) {
+        $member = $this->request("GET", "items/Members/$id");
+        return $this->formatMemberName($member);
+    }
+
+    /** Get list of formatted names from Lassie */
+    public function getListOfNames($active = false): array
     {
         $memberList = $this->getMemberList($active);
         $r = [];
         foreach ($memberList as $member) {
-            if ($member['infix'] !== "") {
-                $r[$member['id']] = $member['first_name'] . " " . $member['infix'] . " " . $member['last_name'];
-            } else {
-                $r[$member['id']] = $member['first_name'] . " " . $member['last_name'];
-            }
+            $r[$member["id"]] = $this->formatMemberName($member);
         }
         return $r;
     }
 
-    /**
-     * Authenticates a member with their ID and password, as registered in Lassie
-     */
-    public function authenticate($id, $pass)
-    {
-        $response = $this->guzzleClient->request('GET', 'api/v2/model', [
-        'query' => ['model_name' => 'person_model', 'method_name' => "get_populated_person", "person_id" => (string) $id]
-    ]);
-
-        $arr = json_decode($response->getBody(), true);
-        $this->checkForError($arr);
-        $hash = $arr['security_hash'];
-        return password_verify($pass, $hash);
+    private static function formatMemberName(array $member) : string{
+        if ($member["infix"] !== "" && $member["infix"] !== null)
+            return $member["first_name"] . " " . $member["infix"] . " " . $member["last_name"];
+        else
+            return $member["first_name"] . " " . $member["last_name"];
     }
 
-    /**
-     * Checks if a password has been registered for this user
-     */
-    public function hasToSetPassword($id)
+    private function getSecurityHash(string $member_id): string
     {
-        $response = $this->guzzleClient->request('GET', 'api/v2/model', [
-            'query' => ['model_name' => 'person_model', 'method_name' => "get_populated_person", "person_id" => (string) $id]
-        ]);
-
-        $arr = json_decode($response->getBody(), true);
-        $this->checkForError($arr);
-        $hash = $arr['security_hash'];
-        if ($hash === "") {
-            return true;
-        }
-        return false;
+        $arr = $this->request("SEARCH", "items/PinHashes",
```

Comment:
Why use SEARCH here? I would just GET request the model for that user and then check the hash. I don't know if this method is compatible with different field types, and SEARCH tends to cause a lot of issues as it's not a standard HTTP method.

Reply:
Well, I made the `PinHash` a separate table from `Member`, so we have to query the `PinHash` table. The supported way to do that in Directus is with a `SEARCH` request.

In order to do as you suggest the `PinHash` should be merged into the `Member` table, which would not be my preference.